### ISSUE TITLE
Have Install-CService default to Local System account

### ISSUE THREAD

**Issue by MDawg957** [webmd-health-services/Carbon#140]
I know when you're running Install-CService if you don't specify the username parameter, it sets it to Network Service but is there a way for it to just default to Local System as shown? Using -Username SYSTEM is mentioned in the documentation, but is there not a way to just have it use the default setting instead of populating it? I'm honestly just trying to edit my service's recovery options, but it doesn't seem like there is a way to change it without using Install-CService. If that's possible, I'll take as as an answer though. Thank you.

No username set: ![image](https://github.com/webmd-health-services/Carbon/assets/12445509/53b249fb-0f66-47d7-91e8-0443bb995a6b)
Service before making changes: ![image](https://github.com/webmd-health-services/Carbon/assets/12445509/b7c7050c-876d-4374-8be1-2cff15c9cf8e)
Username SYSTEM: ![image](https://github.com/webmd-health-services/Carbon/assets/12445509/73327caf-e64a-4311-baa7-da16b5538900)

**Comment by splatteredbits** [comment]
Can you share your code that's calling `Install-CService` ? Are you wanting to use `Install-CService` just to set the recovery options?

**Comment by MDawg957** [comment]
> Can you share your code that's calling `Install-CService` ? Are you wanting to use `Install-CService` just to set the recovery options?

`Install-CService -Name "Secure Spooler Service" -DisplayName "Pharos Systems Print Scout Spooler Service" -Path "C:\PROGRA~2\PHAROS~1\PRINTS~1\SPOOLE~1\WINSPO~1.EXE" -EnsureRunning -OnFirstFailure Restart -OnSecondFailure Restart -OnThirdFailure Restart -Dependency RpcSs, EventLog -UserName SYSTEM`

Ideally I'd like to just use it to set my recovery options. I don't need to change anything except those a `Set-CService` offering would be great as a compromise to adjust existing service settings without re-creating them completely.

As it is, having to input all these extra items seems a bit much to make a simple change, but I could not find another way with the module to do it and it still has the username problem I mentioned in my original post. A streamlined option with something like this would be nice:

Before: 
![image](https://github.com/webmd-health-services/Carbon/assets/12445509/8baf1ca1-f9a9-471a-9bc3-f78d4a884b95)
![image](https://github.com/webmd-health-services/Carbon/assets/12445509/60cd4550-ccfc-43cc-87ec-6d6e2bdccc1f)
![image](https://github.com/webmd-health-services/Carbon/assets/12445509/0eb4de39-a738-4cc6-99c9-35d6cb256771)

After:
![image](https://github.com/webmd-health-services/Carbon/assets/12445509/99bfa1d6-bcba-4255-92e9-b926bcfe91af)
![image](https://github.com/webmd-health-services/Carbon/assets/12445509/17b7f8f7-4cf7-41e1-854f-f90e20234959)
![image](https://github.com/webmd-health-services/Carbon/assets/12445509/a04e9488-7a97-47bd-906b-44b3621bbb23)

A streamlined option with something like this would be nice:

`Set-CService -Name "Secure Spooler Service" -OnFirstFailure Restart -OnSecondFailure Restart -OnThirdFailure Restart`

Everything stays the same, I don't have to identify absolutely every part of the service beforehand to avoid issues, just set what I'd like to change.

**Comment by splatteredbits** [comment]
Agreed that a `Set-CService` function would work best. The way `Install-CService` works under the hood makes it somewhat difficult to change it to meet your needs.

**Comment by MDawg957** [comment]
Could this be made into an enhancement request for such a cmdlet? I think that would be what I'm looking for here. It'd be great to have a way to edit services without re-creating them with `Install-CService` as the only method. I ended up using `sc.exe failure` to do what I needed, but something that could do this native to PowerShell would be a preferable option.

**Comment by splatteredbits** [comment]
Yes. I'll leave this issue open. Unfortunately, all the work we do on Carbon comes from what we need from it at our day job. We won't be able to get to this until we have a need to make the changes ourselves or are working in a similar area and can squeeze it in.
